store/src/test/java/org/apache/rocketmq/store/DefaultMessageStoreCleanFilesTest.java

```diff
@@ -0,0 +1,364 @@
+        MappedFileQueue consumeQueue = getMappedFileQueueConsumeQueue();
+        assertEquals(fileCountConsumeQueue, consumeQueue.getMappedFiles().size());
+
+        int expireFileCount = 15;
+        expireFiles(commitLogQueue, expireFileCount);
+
+        // magic code 10 reference to MappedFileQueue#DELETE_FILES_BATCH_MAX
+        for (int a = 1, fileCount = expireFileCount; a <= (int) Math.ceil((double) expireFileCount / 10); a++, fileCount -= 10) {
+            getCleanCommitLogService().run();
+            getCleanConsumeQueueService().run();
+
+            int expectDeletedCount = fileCount >= 10 ? a * 10 : ((a - 1) * 10 + fileCount);
+            assertEquals(fileCountCommitLog - expectDeletedCount, commitLogQueue.getMappedFiles().size());
+
+            int msgCountPerFile = getMsgCountPerConsumeQueueMappedFile();
+            int expectDeleteCountConsumeQueue = (int) Math.floor((double) expectDeletedCount / msgCountPerFile);
+            assertEquals(fileCountConsumeQueue - expectDeleteCountConsumeQueue, consumeQueue.getMappedFiles().size());
+        }
+    }
+
+
+    /**
+     * run with the vm args:
+     * -Drocketmq.broker.diskSpaceWarningLevelRatio=0.1
+     * -Drocketmq.broker.diskSpaceCleanForciblyRatio=0.1
+     * <p>
+     * make sure disk space usage is greater than 10%
+     */
+    @Test
+    public void testDeleteFilesImmediatelyBySpaceFull() throws Exception {
+        String deleteWhen = "04";
+        // the min value of diskMaxUsedSpaceRatio. make sure disk space usage is greater than 10%
+        int diskMaxUsedSpaceRatio = 10;
+
+        messageStore = initMessageStore(deleteWhen, diskMaxUsedSpaceRatio);
+
+        // build and put 55 messages, exactly one message per CommitLog file.
+        buildAndPutMessagesToMessageStore(msgCount);
+
+        // undo comment out the code below, if want to debug this case rather than just run it.
+        // Thread.sleep(1000 * 60 + 100);
+
+        MappedFileQueue commitLogQueue = getMappedFileQueueCommitLog();
+        assertEquals(fileCountCommitLog, commitLogQueue.getMappedFiles().size());
+
+        int fileCountConsumeQueue = getFileCountConsumeQueue();
+        MappedFileQueue consumeQueue = getMappedFileQueueConsumeQueue();
+        assertEquals(fileCountConsumeQueue, consumeQueue.getMappedFiles().size());
+
+        // In this case, there is no need to expire the files.
+        // int expireFileCount = 15;
+        // expireFiles(commitLogQueue, expireFileCount);
+
+        // magic code 10 reference to MappedFileQueue#DELETE_FILES_BATCH_MAX
+        for (int a = 1, fileCount = fileCountCommitLog;
+             a <= (int) Math.ceil((double) fileCountCommitLog / 10) && fileCount >= 10;
+             a++, fileCount -= 10) {
+            getCleanCommitLogService().run();
+            getCleanConsumeQueueService().run();
+
+            assertEquals(fileCountCommitLog - 10 * a, commitLogQueue.getMappedFiles().size());
```

Comment:
Could you please check this test case again?  you can refer to the CI result like this:
```
org.apache.rocketmq.store.DefaultMessageStoreCleanFilesTest
testDeleteFilesImmediatelyBySpaceFull(org.apache.rocketmq.store.DefaultMessageStoreCleanFilesTest)  Time elapsed: 1.355 sec  <<< FAILURE!
java.lang.AssertionError: expected:<45> but was:<55>
	at org.apache.rocketmq.store.DefaultMessageStoreCleanFilesTest.testDeleteFilesImmediatelyBySpaceFull(DefaultMessageStoreCleanFilesTest.java:184)
```